src/streamlink/plugins/olympicchannel.py

```diff
@@ -1,58 +1,68 @@
-import json
 import logging
 import re
+from html import unescape as html_unescape
 from time import time
 from urllib.parse import urljoin, urlparse
 
 from streamlink.plugin import Plugin
 from streamlink.plugin.api import validate
 from streamlink.stream import HLSStream
+from streamlink.utils import parse_json
 
 log = logging.getLogger(__name__)
 
 
 class OlympicChannel(Plugin):
-    _url_re = re.compile(r"https?://(\w+\.)olympicchannel.com/../(?P<type>live|video|original-series|films)/?(?:\w?|[-\w]+)")
-    _tokenizationApiDomainUrl = """"tokenizationApiDomainUrl" content="/OcsTokenization/api/v1/tokenizedUrl">"""
-    _live_api_path = "/OcsTokenization/api/v1/tokenizedUrl?url={url}&domain={netloc}&_ts={time}"
-
+    _url_re = re.compile(r"https?://(\w+\.)?(?:olympics|olympicchannel)\.com/(?:[\w-]+/)?../.+")
+    _token_api_path = "/tokenGenerator?url={url}&domain={netloc}&_ts={time}"
     _api_schema = validate.Schema(
-        validate.text,
-        validate.transform(lambda v: json.loads(v)),
-        validate.url()
+        validate.transform(parse_json),
+        [{
+            validate.optional("src"): validate.url(),
+            validate.optional("srcType"): "HLS",
+        }],
+        validate.transform(lambda v: v[-1].get("src")),
+    )
+    _data_url_re = re.compile(r'data-content-url="([^"]+)"')
+    _data_content_re = re.compile(r'data-d3vp-plugin="THEOplayer"\s*data-content="([^"]+)"')
+    _data_content_schema = validate.Schema(
+        validate.any(
+            validate.all(
+                validate.transform(_data_url_re.search),
+                validate.get(1),
+            ),
+            validate.all(
+                validate.transform(_data_content_re.search),
+                validate.get(1),
+            ),
+        ),
+        validate.transform(html_unescape),
```

Comment:
It might be better to use `itertags` here instead of having a regex that depends on the attribute order, but it's not really important.

Be aware though that the validation schema does not handle invalid/missing data and will raise an error if none of the regexes match.